### PR TITLE
[ace] Upgrade to ACE 7.0.5, added myself as maintainer

### PR DESCRIPTION
**Pull request by jwillemsen**
**Describe the pull request**

- #### What does your PR fix?  
  Upgrade to ACE 7.0.5

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  `<same as before>, <Yes>`

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  <Yes>

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
@jwillemsen ,Thanks for your pr, I tested all features in x64-windows triplet locally but failed, Could you please take a look:
[build-x64-windows-rel-out.log](https://github.com/microsoft/vcpkg/files/7526757/build-x64-windows-rel-out.log)

**Comment by jwillemsen** [comment]
The error I see is `fatal error C1090: PDB API call failed, error code '23'` but a google search hints at using the latest Visual Studio 2019 release, which visual studio version are you using?

**Comment by JonLiu1993** [comment]
> The error I see is `fatal error C1090: PDB API call failed, error code '23'` but a google search hints at using the latest Visual Studio 2019 release, which visual studio version are you using?

Visual Studio 2019

**Comment by jwillemsen** [comment]
But which update, the latest is 16.11.6, see https://docs.microsoft.com/en-us/visualstudio/releases/2019/release-notes#whats-new-in-visual-studio-2019-version-1611

**Comment by JonLiu1993** [comment]
> But which update, the latest is 16.11.6, see https://docs.microsoft.com/en-us/visualstudio/releases/2019/release-notes#whats-new-in-visual-studio-2019-version-1611

visual-studio-2019 16.10.4

**Comment by jwillemsen** [comment]
I think you hit a compiler bug, did see more references on the web about this, can you try the latest visual studio release?

**Comment by JonLiu1993** [comment]
> I think you hit a compiler bug, did see more references on the web about this, can you try the latest visual studio release?

Thanks @jwillemsen ,I will update my Visual Studio version and test again.

File: ports/ace/portfile.cmake [hunk 1]
```diff
@@ -88,6 +88,7 @@ endif()
 
 # Add ace/config.h file
 # see https://htmlpreview.github.io/?https://github.com/DOCGroup/ACE_TAO/blob/master/ACE/ACE-INSTALL.html
+# ACE 7.0.5 will have support for vs2022
```
Comment (mitza-oci) [inline ports/ace/portfile.cmake:91]: If the current CMake script is used with vs2022 it will generate vc14 projects (due to the `else()` below).  Should this be changed to vs2019?

Comment (jwillemsen) [reply ports/ace/portfile.cmake:91]: With 7.0.5 we should update this file

Comment (mitza-oci) [reply ports/ace/portfile.cmake:91]: I was only commenting on what will happen if the current 7.0.4 version is used (if this PR is merged and a user tries it with VS2022).

Comment (jwillemsen) [reply ports/ace/portfile.cmake:91]: Yes, but using an unsupported compiler version is asking for problems, hopefully we can release next week

Comment (Osyotr) [reply ports/ace/portfile.cmake:91]: > fatal error C1090: PDB API call failed, error code '23'

Perhaps /Zi is used somewhere

**Comment by Osyotr** [comment]
> fatal error C1090: PDB API call failed, error code '23'

Perhaps /Zi is used somewhere.

File: ports/ace/portfile.cmake (hunk 1 above)
Comment (mitza-oci) [reply ports/ace/portfile.cmake:91]: Does that mean this PR is blocked on 7.0.5?

Comment (jwillemsen) [reply ports/ace/portfile.cmake:91]: No, just a reminder that we need to update that part with 7.0.5

Comment (mitza-oci) [reply ports/ace/portfile.cmake:91]: If a user has VS2022 installed and gets the version of vcpkg with this merged, it will not work.  vcpkg will default to 2022's toolset but the projects will be generated for vc14.  If you make a 2-line change the to CMake script you can have it generate vs2019 projects or error out.

File: ports/ace/portfile.cmake [hunk 2]
```diff
@@ -89,7 +89,9 @@ endif()
 # Add ace/config.h file
 # see https://htmlpreview.github.io/?https://github.com/DOCGroup/ACE_TAO/blob/master/ACE/ACE-INSTALL.html
 if(VCPKG_TARGET_IS_WINDOWS)
-  if(VCPKG_PLATFORM_TOOLSET MATCHES "v142")
+  if(VCPKG_PLATFORM_TOOLSET MATCHES "v143")
```
Comment (mitza-oci) [inline ports/ace/portfile.cmake:92]: We can do this in a separate PR, but I think it's more maintainable to check the old versions explicitly and let the unknown Toolset version act like the newest:
```
  if(VCPKG_PLATFORM_TOOLSET MATCHES "v140")
    set(SOLUTION_TYPE vc14)
  elseif(VCPKG_PLATFORM_TOOLSET MATCHES "v141")
    set(SOLUTION_TYPE vs2017)
  elseif(VCPKG_PLATFORM_TOOLSET MATCHES "v142")
    set(SOLUTION_TYPE vs2019)
  else()
    set(SOLUTION_TYPE vs2022)
  endif()
```

Comment (jwillemsen) [reply ports/ace/portfile.cmake]: Done

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Comment by JonLiu1993** [comment]
Thanks everyone for your contribution

**Comment by vicroms** [comment]
Thanks for the PR!
